proposals/0033-data-collections.md

```diff
@@ -0,0 +1,244 @@
+  schema: z.object({
+    name: z.string(),
+    twitter: z.string().url(),
+  })
+});
+
+export const collections = { authors };
+```
+
+These collections can also be queried using the `getCollection()` and `getEntry()` utilities:
+
+```astro
+---
+import { getCollection, getEntry } from 'astro:content';
+
+const authors = await getCollection('authors');
+const ben = await getEntry('authors', 'ben');
+---
+
+<p>
+  Authors: {authors.map(author => author.data.name).join(', ')}
+</p>
+<p>
+  The coolest author: <a href={ben.data.twitter}>{ben.data.name}</a>
+</p>
+```
+
+# Background & Motivation
+
+Content collections are restricted to supporting `.md`, `.mdx`, and `.mdoc` files. This is limiting for other forms of data you may need to store, namely raw data formats like JSON or YAML.
+
+Taking a blog post as the example, there will likely be author information thats reused across multiple blog posts. To standardize updates when, say, updating an author's profile picture, it's best to store authors in a standalone data entry.
+
+The content collections API was built generically to support this future, choosing format-agnostic naming like `data` instead of `frontmatter` and `body` instead of `rawContent`. Because of this, expanding support to new data formats without API changes is a natural progression.
+
+# Goals
+
+- **Introduce JSON and YAML collection support,** configurable and queryable with similar APIs to content collections.
+- **Determine where data collections are stored.** We may allow data collections within `src/content/`, or introduce a new reserved directory.
+
+# Non-Goals
+
+- **Separate RFC:** Referencing data collection entries from existing content collections. This unlocks referencing, say, an author from your blog post frontmatter. See the [related collection references RFC](https://github.com/withastro/roadmap/blob/d89e2a4c28379108501aa6bf40d2f8d93d81ad02/proposals/0034-collection-references.md) for details.
+
+# Detailed Design
+
+Data collections are created with the `defineCollection()` utility, and **must** include `type: 'data'` to store JSON or YAML files. This means "mixed" collections containing both content and data entries are not supported, and should raise a helpful error to correct.
+
+```ts
+// src/content/config.ts
+import { defineCollection, z } from 'astro:content';
+
+const authors = defineCollection({
+  type: 'data',
+  schema: z.object({
+    name: z.string(),
+    twitter: z.string().url(),
+  })
+});
+
+const blog = defineCollection({
+  // `type` can be omitted for content collections
+  // to make this change non-breaking.
+  // `type: 'content'` can also be used for completeness.
+  schema: z.object({...}),
+})
+
+export const collections = { authors };
+```
+
+## Return type
+
+Data collection entries include the same `id`, `data`, and `collection` properties as content collections:
+
+- `id (string)` - The entry file name with the extension omitted. Spaces and capitalization are preserved.
```

Comment:
`id` is the name with extension omitted, does the RFC addresses cases where two files have the same name but **different extensions**?

```
src/content/blog/post.json
src/content/blog/post.yaml
```

Reply:
This should raise an error! Can clarify in the PR. Couldn't find a compelling reason to support this case.